app/src/main/java/com/babylon/wallet/android/presentation/settings/accountsecurity/ledgerhardwarewallets/AddLedgerDeviceViewModel.kt

```diff
@@ -137,3 +152,12 @@ data class AddLedgerDeviceUiState(
         )
     }
 }
+
+sealed interface ShowLinkConnectorPromptState {
+    data object None : ShowLinkConnectorPromptState
+    data class Show(val source: Source) : ShowLinkConnectorPromptState
+
+    enum class Source {
+        AddLedgerDevice, UseLedger
+    }
+}
```

Comment:
this state seems to be redundant. At least in the `LedgerHardwareWalletsViewModel` it is not needed which I proposed a much simpler ui state.

However if you plan to keep it for the other viewmodels, then please move it to the `AddLinkConnectorViewModel`. 

These two seperate viewmodels `AddLedgerDeviceViewModel`, `AddLinkConnectorViewModel` have been created in a way to not be mixed to each other. The add-ledger-logic-state goes to the `AddLedgerDeviceViewModel` and the add-link-connector-logic-state goes to `AddLinkConnectorViewModel`

Reply:
maybe they were, but listening for CE connection status and re-linking mixes all view models together and I don't think I can reduce state as you wrote above